h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -2339,6 +2359,41 @@ else if (gaussian.equals(_parms._family) && Link.identity.equals(_parms._link))
       }
     }
 
+    /***
+     * Estimate dispersion factor using maximum likelihood.  I followed section IV of the doc in 
+     * https://h2oai.atlassian.net/browse/PUBDEV-8683 . 
+     */
+    public double estimateMLSE(GLMTask.ComputeMLSETsk mlCT, double alpha, double[] beta) {
+      double constantValue = mlCT._wsum + mlCT._sumlnyiOui - mlCT._sumyiOverui;
+      DataInfo dinfo = _state.activeData();
+      Frame adaptedF = dinfo._adaptedFrame;
+      int count = 0;
+      long currTime = System.currentTimeMillis();
+      long modelBuiltTime = currTime - _model._output._start_time;
+      while (true) {
+        GLMTask.ComputeDiTriGammaTsk ditrigammatsk = new GLMTask.ComputeDiTriGammaTsk(null, dinfo, _job._key, beta,
+                _parms, alpha).doAll(adaptedF);
+        double numerator = mlCT._wsum*Math.log(alpha)-ditrigammatsk._sumDigamma+constantValue; // equation 2 of doc
+        double denominator = mlCT._wsum/alpha - ditrigammatsk._sumTrigamma;  // equation 3 of doc
+        double change = numerator/denominator;
+        if (denominator == 0 || Double.isNaN(change))
+          return alpha;
+        if (Math.abs(change) < EPS)
+          return alpha-change;
+        else {
+          count++;
+          double se = alpha - change;
+          if (se < 0) // heuristc to prevent seInit <= 0
+            alpha *= 0.5;
+          else
+            alpha = se;
+        }
+        // stop if timeout occurred or if time to find alpha exceeds model building time
```

Comment:
the time-based stopping introduces non-determinism - which is what we usually want to avoid

it is also not clear that the time-based stopping was triggered - I would remove this part completely OR turn it into a warning

Reply:
I added this here to make sure we won't go into in infinite loop.  I limit the time taken to find the best dispersion parameter to be equal to the time it takes to build the GLM model.